pages/1.13/security/ent/tls-ssl/exhibitor/index.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: layout.pug
+navigationTitle:  Securing Exhibitor with mutual TLS
+title: Securing Exhibitor with mutual TLS
+menuWeight: 500
+excerpt: Securing DC/OS with a TLS enabled Exhibitor ensemble
+enterprise: true
+---
+<!-- The source repository for this topic is https://github.com/dcos/dcos-docs-site -->
+
+By default, the Exhibitor HTTP service is open to any client that can reach port 8181 on a master node. This documentation chapter describes a method for protecting the Exhibitor service from unauthorized access. Once enabled, HTTP clients must access Exhibitor through Admin Router; thus applying the Admin Router access control policy to the exhibitor service.
+
+*Note: When accessing Exhibitor through Admin Router (**https://master_host/exhibitor**), authenticated users must have the `dcos:adminrouter:ops:exhibitor` privilege*
+
+# Securing Exhibitor
+
+The strategy for securing exhibitor is mutual TLS authentication. In order to secure exhibitor we must first create a unique root CA. This CA must be used to sign a client certificate for each master node. Creating a public key infrastructure that outputs PEM and Java Keytool formatted artifacts is not a trivial task. To make this processes easier, a simple tool has been created for producing the necessary files. 
+
+*Note: This guide assumes a static exhibitor ensemble. Dynamic masters are not currently supported*
+
+## Using the tool
+
+*Note: A working docker installation is required. If docker is not available see https://github.com/mesosphere/exhibitor-tls-artifacts-gen/blob/master/README.md for information on running the command natively.*
+
+Download the script from the github release page and run it:
+
+```sh
+curl -LsO https://github.com/mesosphere/exhibitor-tls-artifacts-gen/releases/download/0.0.3/exhibitor-tls-artifacts
+chmod +x exhibitor-tls-artifacts
+./exhibitor-tls-artifacts --help
+```
+
+You should see something like this:
+
+    Usage: exhibitor-tls-artifacts [OPTIONS] [NODES]...
+
+    Generates Admin Router and Exhibitor TLS artifacts. NODES should consist
+    of a space separated list of master ip addresses. See
+    https://docs.mesosphere.com/1.13/security/ent/tls-ssl/exhibitor/
+
+    Options:
+    -d, --output-directory TEXT  Directory to put artifacts in. This
+                                 output_directory must not exist.
+    --help                       Show this message and exit.
+
+
+To generate the TLS artifacts, run the tool and provide your master nodes ip addresses as positional arguments. For example, if your master nodes are `10.192.0.2, 10.192.0.3, 10.192.0.4`, invoke the command using:
```

Comment:
>  your master nodes ip addresses as positional arguments

I smell a pitfall here where users might not provide the correct IP addresses.

We should probably make this more precise.

Can we express this in terms of `/opt/mesosphere/bin/detect_ip`?

From a running cluster, on a master node:
```
$ /opt/mesosphere/bin/detect_ip
172.31.6.21
$ /opt/mesosphere/bin/detect_ip_public
34.204.5.37
```

We want the _internal_ IP addresses, i.e. what `/opt/mesosphere/bin/detect_ip` returns on the individual master nodes.